share/plugin/ambit.cc.template

```diff
@@ -0,0 +1,179 @@
+/*
+ *@BEGIN LICENSE
+ *
+ * @plugin@ by Psi4 Developer, a plugin to:
+ *
+ * PSI4: an ab initio quantum chemistry software package
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *@END LICENSE
+ */
+
+#include <psi4-dec.h>
+#include <libparallel/parallel.h>
+#include <liboptions/liboptions.h>
+#include <libmints/mints.h>
+#include <libpsio/psio.h>
+
+#ifndef HAVE_AMBIT
+#error PSI4 was not compiled with Ambit support.
+#endif
+
+#include <ambit/tensor.h>
+#include <ambit/helpers/psi4/integrals.h>
+#include <ambit/helpers/psi4/convert.h>
+
+namespace psi{ namespace @plugin@ {
+
+extern "C"
+int read_options(std::string name, Options &options)
+{
+    if (name == "@PLUGIN@"|| options.read_globals()) {
+        /*- The amount of information printed
+            to the output file -*/
+        options.add_int("PRINT", 1);
+        /*- Whether to compute two-electron integrals -*/
+        options.add_bool("DO_TEI", true);
+    }
+
+    return true;
+}
+
+extern "C"
+SharedWavefunction @plugin@(SharedWavefunction ref_wfn, Options &options)
+{
+    using namespace ambit;
+
+    boost::shared_ptr<Molecule> molecule = ref_wfn->molecule();
+
+    // Form basis object:
+    // Create a basis set parser object.
+    boost::shared_ptr<BasisSetParser> parser(new Gaussian94BasisSetParser());
+    // Construct a new basis set.
+    boost::shared_ptr<BasisSet> aoBasis = BasisSet::pyconstruct_orbital(molecule,
+                                             "BASIS", options.get_str("BASIS"));
```

Comment:
I would stress that we should get orbital basis sets from wavefunctions and not rebuild from the options object.
`boost::shared_ptr<BasisSet> aoBasis = ref_wfn->basisset();`